maven-surefire-plugin/src/main/java/org/apache/maven/plugin/surefire/SurefirePlugin.java

```diff
@@ -880,4 +900,20 @@ protected final ForkNodeFactory getForkNode()
     {
         return forkNode;
     }
+
+    @Override
+    boolean verifyParameters() throws MojoFailureException, MojoExecutionException
```

Comment:
This must not exist in both Plugin classes. See the super method in AbstractSurefireMojo and add a new warn* method and a call which finally calls the abstract method getFailOnFlakeCount.

Reply:
SurefirePlugin and VerifyMojo do not have a common base class.

Reply:
I am not talking about VerifyMojo, nothing but the SurefirePlugin class.

Reply:
But why would the super class contain logic that only applies to a single subclass?

Reply:
Nevermind, I see the if-else in that method now.

Reply:
> 
> 
> But why would the super class contain logic that only applies to a single subclass?

Both plugins, Surefire and Failsafe, have 90% same parameters. So they always contained the method `verifyParameters()` in the super class. The `VerifyMojo` is an exception and it has never had so many config params.